### PR TITLE
#159965295 Configure Database and set up environment variable

### PR DESCRIPTION
**Pull request by mikenthiwa**
What does this PR do?
  * Create environment setting and configure AuthorsHaven app to use PostgreSQL as DB Engine.

Description of Task to be completed?
  * Using `Twelve-factor methodology` to configure the application with environment variables.
  * Install Psycopg2-binary.

Background context :
  * Changing database default settings to read from .env file:
      .env sample file : [sample .env file](https://github.com/andela/ah-magnificent6/blob/ch-configureDB-159965295/.env-sample)
      `DATABASES = {
    'default': env.db()
}`

What are the relevant pivotal tracker stories?
  * [Configure Authors Haven to use PostgreSQL as DB Engine](https://www.pivotaltracker.com/story/show/159965295)

### CODE REVIEW

File: authors/settings.py [hunk 1]
```diff
@@ -11,6 +11,12 @@
 """
 
 import os
+import environ
+import dj_database_url
```
Comment (meshack-mbuvi) [inline authors/settings.py]: I have noted that you have installed and imported dj_database_url but you are not using it.

Comment (mikenthiwa) [reply authors/settings.py]: Noted and rectified

**Comment by Bileonaire** [comment]
I have checked out the changes made. Good work on setting up the environment variables(.env file).

 You can create a  .env_sample file for us to be able to use.

**Review by parseen254 — COMMENTED** [review]
@Bileonaire the env-sample file does not need to be a bash file.....

File: authors/settings.py [hunk 2]
```diff
@@ -11,6 +11,12 @@
...
+
+root = environ.Path(__file__) - 2  # three folder back (/a/b/c/ - 3 = /)
```
Comment (pngg) [inline authors/settings.py]: I've noticed that the ROOT path has been defined twice. ie BASE_DIR and this 'root' variable. Could we please refactor and use the BASE_DIR instead? Also, let's define such constants with all caps.

**Comment by parseen254** [comment]
Also adhere to this while naming your branches and PR's
 https://github.com/andela/bestpractices/wiki/Git-naming-conventions-and-best-practices

**Comment by pabishai** [comment]
@mikenthiwa Consider adding a way to manually test it. I cannot assert that the database connection works as it is.

**Comment by micahoriaso** [comment]
@mikenthiwa please let us not create pull requests that merge into master

**Comment by mikenthiwa** [comment]
I have implemented the required changes.

File: authors/settings.py [hunk 3]
```diff
@@ -80,10 +86,7 @@
 # https://docs.djangoproject.com/en/1.11/ref/settings/#databases
 
 DATABASES = {
-    'default': {
-        'ENGINE': 'django.db.backends.sqlite3',
-        'NAME': os.path.join(BASE_DIR, 'db.sqlite3'),
-    }
+    'default': env.db()
```
Comment (parseen254) [inline authors/settings.py:89]: Are you sure this will work?

**Review by parseen254 — CHANGES_REQUESTED** [review]
@mikenthiwa why does the env-sample file have to be a bash script?

File: authors/settings.py (hunk 3 above)
Comment (mikenthiwa) [reply authors/settings.py:89]: I have disscussed with my peer and he encountered a problem when using the bash script.
I will use .env
I am pushing the changes immediately

**Review by pngg — APPROVED** [review]
The settings seem ok now.

**Review by parseen254 — APPROVED** [review]

**Pull request merged**